tests/iio_genxml.c

```diff
@@ -82,40 +84,37 @@ int main(int argc, char **argv)
 				return EXIT_FAILURE;
 			}
 			backend = NETWORK;
-			arg_index += 2;
-			ip_index = arg_index;
+			arg_ip = optarg;
 			break;
 		case 'x':
 			if (backend != LOCAL) {
 				fprintf(stderr, "-x and -n are mutually exclusive\n");
 				return EXIT_FAILURE;
 			}
 			backend = XML;
-			arg_index += 2;
-			xml_index = arg_index;
+			arg_xml = optarg;
 			break;
 		case 'u':
-			arg_index += 2;
-			uri_index = arg_index;
+			arg_uri = optarg;
 			backend = AUTO;
 			break;
 		case '?':
 			return EXIT_FAILURE;
 		}
 	}
 
-	if (arg_index >= argc) {
+	if (optind != argc) {
```

Comment:
curios: `optind + 1 != argc` here ?

Reply:
optind should be == argc if all arguments have been consumed. It think this is correct the way it is.

Reply:
ack ; thanks for the info ;
i admit i got a bit confused in `optind`  stuff ;